packages/snjs/lib/services/features_service.ts

```diff
@@ -239,58 +243,33 @@ export class SNFeaturesService extends PureService<void> {
         return false;
       });
 
-      switch (feature.content_type) {
-        case ContentType.Component:
-          if (existingItem) {
-            await this.itemManager.changeComponent(
-              existingItem.uuid,
-              (mutator) => {
-                mutator.setContent({
-                  ...existingItem.safeContent,
-                  ...itemContent,
-                });
-              }
-            );
-            if (expired) {
-              this.componentManager.setReadonlyStateForComponent(
-                existingItem as SNComponent,
-                expired
-              );
-            }
-          } else {
-            const newItem = await this.itemManager.createItem(
-              feature.content_type,
-              itemContent
-            );
-            if (expired) {
-              this.componentManager.setReadonlyStateForComponent(
-                newItem as SNComponent,
-                expired
-              );
-            }
-          }
-          break;
-        default:
-          if (existingItem) {
-            await this.itemManager.changeComponent(
-              existingItem.uuid,
-              (mutator) => {
-                mutator.setContent({
-                  ...existingItem.safeContent,
-                  ...itemContent,
-                });
-              }
-            );
-            if (expired) {
-              itemsToDeleteUuids.push(existingItem.uuid);
-            }
-          } else if (!expired) {
-            await this.itemManager.createItem(
-              feature.content_type,
-              itemContent
-            );
+      let resultingItem: SNComponent | undefined = existingItem as SNComponent;
+
+      if (existingItem) {
+        resultingItem = await this.itemManager.changeComponent(
+          existingItem.uuid,
+          (mutator) => {
+            mutator.hosted_url = feature.url;
+            mutator.package_info = feature as ComponentPackageInfo;
+            mutator.valid_until = new Date(feature.expires_at || 0);
           }
-          break;
+        );
+      } else if (!expired || feature.content_type === ContentType.Component) {
```

Comment:
If I'm following correctly with this logic we won't be creating items for inexistent themes, we'd probably need to remove the second condition here?